backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/LocalStorageSynchronizer.java

```diff
@@ -79,7 +78,7 @@ public AnalyzerSettingsUpdateSummary synchronizeAnalyzerConfig(ServerApi serverA
       var originalAnalyzerConfiguration = storage.project(projectKey).analyzerConfiguration().read();
       configUpdateSummary = diffAnalyzerConfiguration(originalAnalyzerConfiguration, updatedAnalyzerConfiguration);
     } catch (StorageException e) {
-      configUpdateSummary = null;
+      configUpdateSummary = new AnalyzerSettingsUpdateSummary(updatedAnalyzerConfiguration.getSettings().getAll());
```

Comment:
As this `synchronizeAnalyzerConfig` method always returns non-null, you can also remove the null check in `SynchronizationService`.